spec/controllers/videos_controller_spec.rb

```diff
@@ -0,0 +1,28 @@
+RSpec.describe VideosController, type: :controller do
+  describe 'GET show' do
+    context 'when a video does exist' do
+      let(:categories) { [] }
+      let(:video) { Core::Video.new('test', title: 'test', body: 'body', categories: categories) }
+
+      before do
+        expect(Core::VideoReader).to receive(:new) do
+          double(Core::VideoReader, call: video)
+        end
+      end
+
+      it 'is successful' do
+        get :show, id: video.id, locale: I18n.locale
+        expect(response).to be_ok
+      end
+    end
+
+    context 'when an video does not exist' do
+      before { allow_any_instance_of(Core::VideoReader).to receive(:call).and_yield }
```

Comment:
I tend to double take when I see `allow_any_instance_of` it's usually considered a smell. Can we avoid it here?